documentation/src/main/asciidoc/mapper-orm-indexing.asciidoc

```diff
@@ -1,41 +1,144 @@
+Even when you change only a single property of an indexed entity,
+if that property is indexed,
+Hibernate Search needs to rebuild the corresponding document *in full*.
++
+Even if Hibernate Search tries to only load what is necessary for indexing,
+depending on your mapping, this may lead to lazy associations being loaded just to reindex entities,
+even if you didn't need them in your business code.
++
+This extra cost can be mitigated to some extent by leveraging Hibernate ORM's batch fetching;
+see link:{hibernateDocUrl}#configurations-database-fetch[the `batch_fetch_size` property]
+and link:{hibernateDocUrl}#fetching-batch[the `@BatchSize` annotation].
 
-* by default, or when set to `true`, Hibernate Search will only trigger reindexing if the properties of the entity that changed
-are actually used as a source to generate the indexable document.
-* when set to `false`, Hibernate Search will trigger reindexing regardless of the entity properties that changed.
-
-[[mapper-orm-indexing-automatic]]
-== Automatic indexing
+Automatic indexing may be unnecessary if your index is read-only
+or if you update it regularly through <<mapper-orm-indexing-explicit,explicit indexing>>.
+You can enable or disable automatic indexing by setting the configuration property
+`hibernate.search.automatic_indexing.strategy`:
 
-include::todo-placeholder.asciidoc[]
+* when set to `session`, each change to an indexed entity
+(persist, update, delete) through a Hibernate ORM Session/EntityManager
+will automatically lead to a similar modification to the index.
+* when set to `none`, changes to entities are ignored,
+and indexing requires an explicit action (see <<mapper-orm-indexing-explicit>>).
 
 [[mapper-orm-indexing-automatic-synchronization]]
-=== Synchronization with the index
+=== Synchronization with the indexes
 
 include::todo-placeholder.asciidoc[]
+// TODO HSEARCH-3317 explain in transaction/post transaction synchronizations, when they can be used and what they mean
+// TODO HSEARCH-3305 explain explicit flush/refresh
+
+Hibernate Search offers multiple strategies to control synchronization with the indexes
+during automatic indexing,
+i.e. to control the minimum progress of indexing before the transaction commit returns.
+
+You can define a default strategy by setting the configuration property
+`hibernate.search.automatic_indexing.synchronization_strategy`:
+
+* when set to `queued`, the transaction commit will return as soon as
+the index changes are queued in the backend.
++
+This strategy offers no guarantee as to whether indexing will be performed successfully,
+or even whether indexing will be performed at all:
+the local JVM may crash before the works are executed, in which case the indexing requests will be forgotten,
+or indexing may simply fail.
+* by default or when set to `committed`, the transaction commit will return as soon as
+the index changes are committed to disk.
++
+This generally means that at the very least
```

Comment:
Maybe we can say that if a remote server is involved to store the indexes, this strategy guarantees that we've received a positive feedback from it.

Reply:
That's what this sentence means, but since it's not clear I'll add something.